### PR TITLE
Add asarray to explicitly cast list to numpy array in wilcoxon test

### PR DESCRIPTION
**Pull request by wuhaochen**
An easy fix to address issue #6017

### CODE REVIEW

**Comment by ev-br** [comment]
Can you add the failing example from #6017 as a regression test please.

**Comment by wuhaochen** [comment]
Test cases added.

File: scipy/stats/tests/test_morestats.py [hunk 1]
```diff
@@ -757,6 +757,22 @@ def test_wilcoxon_bad_arg():
...
+    assert_allclose(T, 2.5)
+    assert_allclose(p, 0.091690281549429153)
+
+    T, p = stats.wilcoxon(arr, zero_method="zsplit")
+    assert_allclose(T, 3.0)
+    assert_allclose(p, 0.11384629800665805)
+
+    T, p = stats.wilcoxon(arr, zero_method="wilcox")
+    assert_allclose(T, 1.5)
+    assert_allclose(p, 0.10247043485974937)
```
Comment (ev-br) [inline scipy/stats/tests/test_morestats.py]: 1. where are these numbers from? 
2. please mention the issue number, gh-6070 in the comment. 
3. If the objective of the test is to make sure lists pass through correctly, you may or may not want to test specific numbers.

Comment (wuhaochen) [inline scipy/stats/tests/test_morestats.py]: Is there an established tradition of how to mention the issue number? Shall I include url or just mention gh-6070?

**Comment by ev-br** [comment]
Squashed and merged in fa3bfb902099b1a465b82c71b5bdca34e04c6b00.
Thank you @wuhaochen and congratulations with what I believe is your first scipy commit!
(the commit hash is 6d07744405d628c5db7251c16ebdde718f1cf088)

Next time (and we're looking for your next contributions), it'd be nice to have commit messages in a format recommended in http://docs.scipy.org/doc/numpy-1.10.1/dev/gitwash/development_workflow.html

**Pull request closed without merging**